File: mobile/RhodestheCarnaby/app/Shop/shop.rb
# The model has already been created by the framework, and extends Rhom::RhomObject
# You can add more methods here
class Shop
  # include Rhom::PropertyBag
  include Rhom::FixedSchema

  # Uncomment the following line to enable sync with Brand.
  # enable :sync

  #add model specifc code here
  
  set :schema_version, '0.2' #application may use it for data migration
 
  # object properties 
  property :id, :integer
  property :shop_type, :string  
  property :name, :string
  property :description, :text

  property :address, :string
  property :province, :string
  property :city, :string

  property :name_en, :string
  property :description_en, :text

  property :address_en, :string
  property :province_en, :string
  property :city_en, :string

  # property :coordinate, :string  
  property :latitude, :string
  property :longitude, :string

  property :updated_at, :datetime

  # blob fields called out explicitly to identify field type
  # :blob - declare property as a blob type
  # :overwrite - (optional) overwrite client copy of blob with new copy from server
  #                    useful when server modifies images, for example zooming or cropping
  # property :cropped_image_url, :blob, :overwrite
 
 
  # object column will be added by rhom and  will be primary key
 
  #indexes
  # index :by_name_tag, [:name, :tag] #will create index for name and tag columns
  index :by_shop_id_name, [:id, :name]
 
  # unique_index :by_phone, [:phone] #will create unique index for phone column
  unique_index :by_shop_id, [:id]
end

File: mobile/android/src/com/koocaa/thecarnaby/model/Shop.java
package com.koocaa.thecarnaby.model;

public class Shop {
	public int id;
	public String shop_type, 
		name, description, province, city, address,
		name_en, description_en, province_en, city_en, address_en,
		latitude, longtitude;
	public int image_res_id;
	
	public Shop(){
		
	}
	
//	new Shop(1,"零售", "北京中关村津乐汇","","北京","海淀","北京市海淀区中关村大街15号中关村购物中心D区津乐汇地下二层","Beijing Zhongguancun La Vita","","B2 La Vita, D Area Zhonguancun Shopping Center, No.15 Zhongguancun Street, Haidian District, Beijing","Beijing","Haidian","39.97987","116.316429"),
	public Shop(int id, String shop_type, String name, String description, String province, String city, String address, 
			String name_en, String description_en, String province_en, String city_en, String address_en, String latitude, String longtitude, int image_res_id){
		this.id = id;
		this.shop_type = shop_type;
		this.name = name;
		this.description = description;
		this.province = province;
		this.city = city;
		this.address = address;
		this.name_en = name_en;
		this.description_en = description_en;
		this.province_en = province_en;
		this.city_en = city_en;
		this.address_en = address_en;
		this.latitude = latitude;
		this.longtitude = longtitude;
		this.image_res_id = image_res_id;
	}
}
